samza-core/src/main/java/org/apache/samza/storage/ChangelogPartitionManager.java

```diff
@@ -74,16 +98,76 @@ public void register(TaskName taskName) {
   }
 
   /**
-   * Write the taskName to partition mapping that is being maintained by this ChangelogManager
+   * Write the taskName to partition mapping.
    * @param changelogEntries The entries that needs to be written to the coordinator stream, the map takes the taskName
    *                       and it's corresponding changelog partition.
    */
-  public void writeChangeLogPartitionMapping(Map<TaskName, Integer> changelogEntries) {
+  public void writePartitionMapping(Map<TaskName, Integer> changelogEntries) {
     log.debug("Updating changelog information with: ");
     for (Map.Entry<TaskName, Integer> entry : changelogEntries.entrySet()) {
       log.debug("TaskName: {} to Partition: {}", entry.getKey().getTaskName(), entry.getValue());
       send(new SetChangelogMapping(getSource(), entry.getKey().getTaskName(), entry.getValue()));
     }
   }
 
+  /**
+   * Merge previous and new taskName to partition mapping and write it.
+   * @param prevChangelogEntries The previous map of taskName to changelog partition.
+   * @param newChangelogEntries The new map of taskName to changelog partition.
+   */
+  public void updatePartitionMapping(Map<TaskName, Integer> prevChangelogEntries,
+      Map<TaskName, Integer> newChangelogEntries) {
+    Map<TaskName, Integer> combinedEntries = new HashMap<>(newChangelogEntries);
+    combinedEntries.putAll(prevChangelogEntries);
+    writePartitionMapping(combinedEntries);
+  }
+
+  /**
+   * Create and validate changelog streams
+   * @param config Config with changelog info
+   * @param maxChangeLogStreamPartitions Maximum changelog stream partitions to create
+   */
+  public void createChangeLogStreams(Config config, int maxChangeLogStreamPartitions) {
+    // Get changelog store config
+    JavaStorageConfig storageConfig = new JavaStorageConfig(config);
+    Map<String, SystemStream> storeNameSystemStreamMapping = storageConfig.getStoreNames()
+        .stream()
+        .filter(name -> StringUtils.isNotBlank(storageConfig.getChangelogStream(name)))
+        .collect(Collectors.toMap(name -> name,
+            name -> Util.getSystemStreamFromNames(storageConfig.getChangelogStream(name))));
+
+    // Get SystemAdmin for changelog store's system and attempt to create the stream
+    SystemAdmins systemAdmins = new SystemAdmins(config);
```

Comment:
I think this will load all systems. You might want to load just the changelog system admin.

Reply:
I thought `StorageConfig` just has the changelog systems, but I'll double check.